src/middlewared/middlewared/plugins/pool.py

```diff
@@ -1549,7 +1549,10 @@ async def attachments(self, oid):
         share, asking for confirmation.
         """
         pool = await self.get_instance(oid)
-        return await self.middleware.call('pool.dataset.attachments', pool['name'])
+        if pool['status'] in ('OFFLINE', 'FAULTED'):
```

Comment:
I think we can still have attachments in these states? e.g. shares configured for the pool/dataset that is now gone

Reply:
What is the problem you are trying to solve?

Reply:
@themylogin if the pool is not available or we are not able to get the metadata like list of datasets or if that list is malformed, `pool.dataset.attachments` fails as we try to find the dataset in question and then using that dataset's path we find related attachments etc. As @william-gr pointed out, we can still/should find attachments and I'll be making the necessary amends.